src/network/postgres/postgres_protocol_interpreter.cpp

```diff
@@ -83,7 +84,25 @@ Transition PostgresProtocolInterpreter::ProcessStartup(const common::ManagedPoin
       NETWORK_LOG_TRACE(db_name);
     }
   }
-  auto oids = t_cop->CreateTempNamespace(context->GetConnectionID(), db_name);
+
+  std::pair<catalog::db_oid_t, catalog::namespace_oid_t> oids;
+
+  uint32_t sleep_time = INITIAL_BACKOFF_TIME;
+  bool failed_once = false;
+
+  // we loop with exponential backoff because in case there are multiple other pg connections also starting
+  // at the same time, creating DDL conflicts when creating the temp namespace
+  do {
+    // this is so that we don't sleep on our first attempt
+    if (failed_once) {
+      std::this_thread::sleep_for(std::chrono::milliseconds{sleep_time});
+      sleep_time *= BACKOFF_FACTOR;
+    }
+    oids = t_cop->CreateTempNamespace(context->GetConnectionID(), db_name);
+    failed_once = true;
+  } while ((oids.first == catalog::INVALID_DATABASE_OID || oids.second == catalog::INVALID_NAMESPACE_OID) &&
```

Comment:
I still don't think we want INVALID_DATABASE_OID as a condition here. If the database name is wrong, we're gonna spin in this loop, right?

Reply:
Propose getting rid of `failed_once` and refactoring to:
```
 do {
    oids = t_cop->CreateTempNamespace(context->GetConnectionID(), db_name);
    if (oids.first == catalog::INVALID_DATABASE_OID || oids.second != catalog::INVALID_NAMESPACE_OID) break;
    std::this_thread::sleep_for(std::chrono::milliseconds{sleep_time});
    sleep_time *= BACKOFF_FACTOR;
  } while (sleep_time <= MAX_BACKOFF_TIME);
```
Does that logic seem right?

Reply:
Ya this works.